plugins/pulumi/pulumi_access_token.go

```diff
@@ -109,3 +111,31 @@ func TryPulumiConfigFile() sdk.Importer {
 		}
 	})
 }
+
+// DeleteSecretsFileProvisioner removes the Pulumi credentials file before letting the real provisioner do its magic.
+type DeleteSecretsFileProvisioner struct {
+	sdk.Provisioner
+
+	RealProvisioner sdk.Provisioner
+}
+
+func DeleteSecretsFile(provisioner sdk.Provisioner) sdk.Provisioner {
+	return DeleteSecretsFileProvisioner{
+		RealProvisioner: provisioner,
+	}
+}
+
+func (p DeleteSecretsFileProvisioner) Provision(ctx context.Context, in sdk.ProvisionInput, out *sdk.ProvisionOutput) {
+	// delete the current credentials file.
+	os.Remove(fmt.Sprintf("%s/%s", in.HomeDir, ".pulumi/credentials.json"))
```

Comment:
Perhaps it would be better to run `pulumi logout` rather than unceremoniously deleting the creds file.